pkg/operator/ceph/controllers/clusterdisruption/location.go

```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterdisruption
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/rook/rook/pkg/clusterd"
+	cephClient "github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/osd"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileClusterDisruption) getOsdDataList(request reconcile.Request) ([]OsdData, error) {
+	osdDeploymentList := &appsv1.DeploymentList{}
+	namespaceListOpts := client.InNamespace(request.Namespace)
+	err := r.client.List(context.TODO(), osdDeploymentList, client.MatchingLabels{k8sutil.AppAttr: osd.AppName}, namespaceListOpts)
+	if err != nil {
+		return nil, fmt.Errorf("could not list osd deployments: %+v", err)
+	}
+
+	osds := make([]OsdData, 0)
+
+	for _, deployment := range osdDeploymentList.Items {
+		osdData := OsdData{Deployment: deployment}
+		labels := deployment.Spec.Template.ObjectMeta.GetLabels()
+		osdID, ok := labels[osd.OsdIdLabelKey]
+		if !ok {
+			return nil, fmt.Errorf("osd %s was not labeled with ", deployment.ObjectMeta.Name)
+		}
+		osdIDInt, err := strconv.Atoi(osdID)
+		if err != nil {
+			return nil, fmt.Errorf("osd id not an int: %s", osdID)
```

Comment:
Is this the real error? To me, it's more `cannot convert osd id to an int`.

Reply:
If it's an int, it'll convert right?